### PR TITLE
UCS/VFS: Generic callback to print primitive types

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: src/ucs/vfs/base/vfs_obj.h [hunk 1]
```diff
@@ -9,11 +9,56 @@
...
+/**
+ * Defines a function prototype to fill information about an object to a given
+ * string buffer.
+ *
+ * @param _name  Function name.
+ *
+ * @param [in]    obj      Pointer to the object.
+ * @param [inout] strb     String buffer filled with the object's information.
+ * @param [in]    arg_ptr  Optional pointer argument passed to the function
+ * @param [in]    arg_u64  Optional numeric argument passed to the function
```
Comment (dmitrygx) [inline src/ucs/vfs/base/vfs_obj.h:28]:
```suggestion
 * @param [in]    arg_ptr  Optional pointer argument passed to the function.
 * @param [in]    arg_u64  Optional numeric argument passed to the function.
```

**Review by dmitrygx — APPROVED** [review]

File: src/ucs/vfs/base/vfs_obj.h [hunk 2]
```diff
@@ -9,11 +9,56 @@
...
+ */
+#define UCS_VFS_FILE_SHOW(_name) \
```
Comment (brminich) [inline src/ucs/vfs/base/vfs_obj.h:30]: why need this macro? Can we add just `ucs_vfs_show_primitive`?

Comment (yosefe) [reply src/ucs/vfs/base/vfs_obj.h:30]: it's done because we need to add arg_u64 to all existing callbacks, so might as well make a macro to declare these callbacks (reduce code duplication and simplify changing the callback prototype in the future)

Comment (brminich) [reply src/ucs/vfs/base/vfs_obj.h:30]: imo, this makes code less readable and hard to parse by ctags/cscope/etc. Also we do not use such macro for other callback definitions (ams for instance)

File: src/ucs/vfs/base/vfs_obj.h [hunk 3]
```diff
@@ -9,11 +9,40 @@
...
+    UCS_VFS_TYPE_POINTER,
+    UCS_VFS_TYPE_STRING,
+    UCS_VFS_TYPE_CHAR,
+    UCS_VFS_TYPE_SHORT,
+    UCS_VFS_TYPE_INT,
+    UCS_VFS_TYPE_LONG,
+
+    /* Type modifiers */
+    UCS_VFS_TYPE_UNSIGNED = UCS_BIT(14),
+    UCS_VFS_TYPE_HEX      = UCS_BIT(15),
```
Comment (brminich) [inline src/ucs/vfs/base/vfs_obj.h:31]: minor: I'd start these flags from `UCS_VFS_TYPE_LAST` to avoid potential intersection of types and modifiers in the future

**Review by brminich — APPROVED** [review]

File: src/ucs/vfs/base/vfs_obj.h (hunk 3 above)
Comment (yosefe) [reply src/ucs/vfs/base/vfs_obj.h:32]: added static assert since could not find method to round_pow2 in "compile time"

**Review by brminich — APPROVED** [review]
